### DISCUSSION TITLE
Next release plan?

### DISCUSSION THREAD

**Discussion by jayvdb** [juhaku/utoipa#440] [General]
There have been a lot of improvements since the 2.4.2 release.
I would like to be using https://github.com/juhaku/utoipa/pull/405 , and I am sure others are waiting to use other additions.

What is needed before the next release? i.e. where can others help get involved that will help get the next release out sooner?  Are there any regressions?

https://github.com/juhaku/utoipa/issues/312 is the only issue flagged as a bug, but that pre-dates the latest release so it probably isnt a regression.

**Comment by juhaku** [comment]
> What is needed before the next release? i.e. where can others help get involved that will help get the next release out sooner? Are there any regressions?

Hey good question.  The next release will be a breaking release. But the breaking changes are not massive and will not cause any issues for the most of users (at least to those who use derive macros).

There still is some cleaning up and updating the documentation that needs to be done for the next release. All the features that I have thought to be there should be there by now.  There are no milestones nor tracking of features / updates / fixes that should be in a specific release.  Though creating a board to track a next release would be probably beneficial. That would work as a stickyboard for tasks that should be there and are open to pick up by anyone.

But in anyways all the issues that seem actionable e.g. enhancements are open for anyone to pick. :slightly_smiling_face: And of course the items here in the kanban board as well: https://github.com/users/juhaku/projects/1/views/1

There should not be any regressions at least not known. The open bug item listed in issues should be already sorted out at least it hasn't seen any activity for sometime now so it is probably expected to work now. But further testing might be needed e.g. with a bigger OpenAPI specification set.

And yes the release, it can be expected quite soon, within a few days to 2 weeks depending if no fundamental issues arise that need immediate action.  It has surely been quite some time since the previous release, I have felt quite overwhelmed recently with everything and there was the holidays in between as well.

**Comment by jayvdb** [comment]
Is there a place where release notes are being drafted?

While testing `master` I ran into https://github.com/juhaku/utoipa/pull/378 having removed the "json" feature.  Will only effect people who have used `default-features = false`, but feels to be worth relnoting.

**Reply by juhaku** [reply]
> Is there a place where release notes are being drafted?

Yes the release notes are being drated to the Releases section from the Github git log automatically. But It is good to write a post about breaking changes here in the Discussions. 

Previously there was the post about release 2.0.0 and similarly I will write one for 3.0.0. There is not much breaking changes including the `json` feature you mentioned. Rest of the breaking changes are only changes in OpenAPI types in utoipa. Some return types and function arguments have been changed a bit yet should not affect users who purely rely on macros. But those who have manually implemented OpenAPI types previsously need to prepare for small changes.

**Reply by juhaku** [reply]
Here is the breaking changes of next release https://github.com/juhaku/utoipa/discussions/456.
